File: prime.pl
#!/usr/bin/perl
main();

sub prime {
    for (2..int($_[0]/2)) {
        return 0 if ($_[0] % $_ == 0);
    }
    return 1;
}

sub main() {
    my $num = 0;
    for (2..100000) {
        $num++ if (prime($_));
    }
    print("number of primes = $num\n");
}

File: prime.lua
#!/usr/bin/lua

function prime(num)
   for i = 2, num/2, 1 do
      if num % i == 0 then
         return false
      end
   end
   return true
end

function main()
   num = 0;
   for i = 2, 100000, 1 do
      if prime(i) then
         num = num + 1
      end
   end
   print("number of primes = " .. num);
end

main()
